### PR TITLE
PD-532 PD-571 Split out onetrust

### PR DESCRIPTION
**Pull request by edwardupton**

### CODE REVIEW

File: src/customerPrivacyAPI/oneTrust.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { CustomWindow } from '../..';
+
+declare let window: CustomWindow;
+
+const performanceCookieCategory = '2,';
+
+export const waitForOneTrust = (): void => {
+	const interval = setInterval(function () {
```
Comment (lyykfi) [inline src/customerPrivacyAPI/oneTrust.ts:8]: should we have a deadline for the interval?

File: src/customerPrivacyAPI/trustArc.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	window.Shopify?.customerPrivacy?.setTrackingConsent?.(userCanBeTracked, () => {
+		// no callback needed
+	});
+
+(function () {
+	/*
+	First step is to register with the TrustArc API to recieve callbacks when a preference update occurs
+	You must wait for the TrustArc API (PrivacyManagerAPI object) to exist on the page before registering
+	*/
+	const interval = setInterval(function () {
```
Comment (lyykfi) [inline src/customerPrivacyAPI/trustArc.ts:23]: the same

**Review by lyykfi — CHANGES_REQUESTED** [review]

**Comment by lyykfi** [comment]
@edwardupton tests were waiting to long. I think we can restart them but in future we need to fix tests issues here.

**Comment by lyykfi** [comment]
@edwardupton could you fix codeclimate please?

**Review by lyykfi — APPROVED** [review]

**Comment by edwardupton** [comment]
@lyykfi fixed

**Comment by lyykfi** [comment]
@edwardupton nice! thanks.

File: src/customerPrivacyAPI/oneTrust.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { CustomWindow } from '../..';
```
Comment (lyykfi) [inline src/customerPrivacyAPI/oneTrust.ts:1]: I think this file should be merged already to staging, isn't it?

Comment (edwardupton) [reply src/customerPrivacyAPI/oneTrust.ts:1]: Yes it used elsewhere

**Review by lyykfi — APPROVED** [review]

**Pull request merged**